src/libsystemd/sd-bus/bus-objects.c

```diff
@@ -1204,12 +1227,6 @@ static int process_get_managed_objects(
         if (require_fallback || !n->object_managers)
                 return 0;
 
-        r = get_child_nodes(bus, m->path, n, CHILDREN_RECURSIVE, &s, &error);
-        if (r < 0)
-                return bus_maybe_reply_error(m, r, &error);
-        if (bus->nodes_modified)
-                return 0;
```

Comment:
It's essential call get_child_nodes(), as only that calls the enumerator callbacks, i.e. the callbacks registered with sd_bus_add_node_enumerator(), which together with fallback handlers allow you to generate bus objects on-the-fly without registering each individually beforehand...

I figure in order to implement what you are looking for we'd have to rework this stuff to simply use OrderedSet instead of Set.